### PR TITLE
Random seed fix

### PR DESCRIPTION
**Pull request by jeraud**

### CODE REVIEW

**Review by rhandberg — APPROVED** [review]
Except for the fact that we are using `numpy.random.seed` instead of `numpy.random.RandomState`, which I think is going to be deprecated in some upcoming version of numpy, I think this looks (mostly) harmless.
But since this is messing about in his code, I think we should ask @DJArmstrong to review the changes before merging.

File: starclass/RFGCClassifier/RF_GC.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def __init__(self, clfile='rfgc_classifier_v01.pickle', somfile='rfgc_som.txt',
 			if self.classifier.som is None and self.somfile is not None:
 				# load som
 				if os.path.exists(self.somfile):
-					self.classifier.som = fc.loadSOM(self.somfile)
+					self.classifier.som = fc.loadSOM(self.somfile, self.random_seed)
```
Comment (rhandberg) [inline starclass/RFGCClassifier/RF_GC.py:90]: Change this to a keyword-argument (`random_seed=self.random_seed`)

File: starclass/RFGCClassifier/RF_GC_featcalc.py [hunk 2]
```diff
@@ -120,20 +120,20 @@ def prepLCs(lc, linflatten=False):
...
+	nsteps=300, learningrate=0.1, random_seed=None):
 	"""
 	Top level function for training a SOM.
 	"""
 	logger = logging.getLogger(__name__)
 	logger.info('Preparing lightcurves for SOM')
 	SOMarray = SOM_alldataprep(features, cardinality=cardinality)
 	logger.info('%d lightcurves prepared. Training SOM', SOMarray.shape[0])
-	som = SOM_train(SOMarray, outfile, overwrite, cardinality, dimx, dimy, nsteps, learningrate)
+	som = SOM_train(SOMarray, outfile, overwrite, cardinality, dimx, dimy, nsteps, learningrate, random_seed)
```
Comment (rhandberg) [inline starclass/RFGCClassifier/RF_GC_featcalc.py:131]: Change this to a keyword-argument (`random_seed=self.random_seed`)

**Review by rhandberg — CHANGES_REQUESTED** [review]
Just a couple of small changes

**Comment by DJArmstrong** [comment]
All looks ok to me

**Pull request merged**